packages/opencode/src/session/processor.ts

```
@@ -348,10 +348,27 @@ export namespace SessionProcessor {
+                  // would then false-positive on the final text-only step. Also
+                  // scan streamInput.messages for any prior assistant tool-call
+                  // content; if found, the session has used tools and the warning
+                  // should be suppressed.
+                  const sessionHasPriorToolCalls =
+                    sessionToolCallsMade > 0 ||
+                    streamInput.messages.some(
+                      (m) =>
+                        m.role === "assistant" &&
+                        Array.isArray(m.content) &&
+                        m.content.some((p) => p.type === "tool-call"),
+                    )
```

Biome (2.4.16)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)